src/SourceBuild/tarball/patches/msbuild/0004-Remove-dependency-on-MS.VS.Setup.Configuration.Inter.patch

```diff
@@ -0,0 +1,115 @@
++++ b/src/Build/Microsoft.Build.csproj
+@@ -38,8 +38,10 @@
+     <PackageReference Include="System.Reflection.Metadata" Condition="'$(MonoBuild)' == 'true'" />
+   </ItemGroup>
+ 
+-  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
++  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(DotNetBuildFromSource)' != 'true'">
+     <PackageReference Include="Microsoft.VisualStudio.Setup.Configuration.Interop" />
++  </ItemGroup>
++  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+     <Reference Include="System.IO.Compression" />
+     <PackageReference Include="System.Memory" />
+   </ItemGroup>
+diff --git a/src/Directory.BeforeCommon.targets b/src/Directory.BeforeCommon.targets
+index 913c97b12..320bffa56 100644
+--- a/src/Directory.BeforeCommon.targets
++++ b/src/Directory.BeforeCommon.targets
+@@ -99,7 +99,7 @@
+     <DefineConstants>$(DefineConstants);FEATURE_DEBUGGER</DefineConstants>
+     <DefineConstants Condition="'$(MachineIndependentBuild)' != 'true'">$(DefineConstants);FEATURE_WIN32_REGISTRY</DefineConstants>
+     <DefineConstants>$(DefineConstants);FEATURE_WORKINGSET</DefineConstants>
+-    <DefineConstants Condition="'$(MonoBuild)' != 'true' and '$(MachineIndependentBuild)' != 'true' and '$(TargetFrameworkVersion)' != 'v3.5'">$(DefineConstants);FEATURE_VISUALSTUDIOSETUP</DefineConstants>
++    <DefineConstants Condition="'$(MonoBuild)' != 'true' and '$(MachineIndependentBuild)' != 'true' and '$(TargetFrameworkVersion)' != 'v3.5' and '$(DotNetBuildFromSource)' != 'true'">$(DefineConstants);FEATURE_VISUALSTUDIOSETUP</DefineConstants>
+     <DefineConstants>$(DefineConstants);FEATURE_MSCOREE</DefineConstants>
+   </PropertyGroup>
+ 
+diff --git a/src/Samples/Directory.Build.props b/src/Samples/Directory.Build.props
+index 6b9d6cd9f..317ef14a6 100644
+--- a/src/Samples/Directory.Build.props
++++ b/src/Samples/Directory.Build.props
+@@ -8,6 +8,7 @@
+     <EnableCentralPackageVersions>false</EnableCentralPackageVersions>
+ 
+     <IsShipping>false</IsShipping>
++    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
+   </PropertyGroup>
+ 
+   <!-- Import parent props -->
+diff --git a/src/Samples/Directory.Build.targets b/src/Samples/Directory.Build.targets
+index 78b52aab8..db80eff4d 100644
+--- a/src/Samples/Directory.Build.targets
++++ b/src/Samples/Directory.Build.targets
+@@ -2,7 +2,7 @@
+ 
+   <ItemGroup>
+     <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0" PrivateAssets="all"/>
+-    <PackageReference Include="Microsoft.VisualStudio.SDK.EmbedInteropTypes" Version="15.0.15" PrivateAssets="All" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'" />
++    <PackageReference Include="Microsoft.VisualStudio.SDK.EmbedInteropTypes" Version="15.0.15" PrivateAssets="All" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(DotNetBuildFromSource)' != 'true'" />
```

Comment:
Is this really necessary, with the directory being excluded in the `src/Samples/Directory.Build.props` change?

This seems to be about `Microsoft.VisualStudio.SDK.EmbedInteropTypes` rather than `MS.VS.Setup.Configuration.Interop`, should it go into its own patch?

Reply:
Removed.